From 71a146071d7937e9d18aaff881caefdd3ec5e3e9 Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Wed, 12 Feb 2020 06:25:31 +0200
Subject: [PATCH] Merge pull request #1759 in SW_UGW/linux from
 bugfix/UGW_SW-47012-icc-driver-init-routine-return-error-fix to 8.4.1

* commit 'b6ca7b55dfdcd170a01029ac53ea39b430b42029':
  UGW_SW-47012 return value fix and remove module params
---
 drivers/char/ltq_icc/drv_icc.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/char/ltq_icc/drv_icc.c b/drivers/char/ltq_icc/drv_icc.c
index 6bccaffcce03..d3af42a06c78 100644
--- a/drivers/char/ltq_icc/drv_icc.c
+++ b/drivers/char/ltq_icc/drv_icc.c
@@ -581,7 +581,7 @@ int icc_os_register (void){
 	 }
 		TRACE(ICC, DBG_LEVEL_HIGH, ("Major Id is %d\n",icc_major_id));
    	TRACE(ICC, DBG_LEVEL_HIGH, ("ICC driver registered\n"));
-   return ret;
+   return 0;
 }
 
 /**
@@ -782,8 +782,11 @@ int __init icc_init_module (void){
 #ifdef KTHREAD
 	sema_init(&icc_callback_sem,1);
 #endif
+	return 0;
 finish:
-	return result;
+	mps_unregister_callback();
+	mps_close((struct inode *)1, NULL);
+	return -EINVAL;
 }
 
 /*Exit module routine*/
@@ -972,8 +975,6 @@ int icc_unregister_callback (icc_devices type)
 /*Module related definitions*/
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Swaroop Sarma");
-MODULE_PARM_DESC (icc_major_id, "Major ID of device");
-module_param (icc_major_id, short, 0);
 /*module functions of ICC driver*/
 module_init (icc_init_module);
 module_exit (icc_cleanup_module);
